modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/opentofu/opentofulocal/OpenTofuLocalExecutor.java

```diff
@@ -192,8 +193,17 @@ public void destroy() {
      */
     public String getTerraformState() {
         String state = null;
+        String path = workspace + File.separator + STATE_FILE_NAME;
         try {
-            File tfState = new File(workspace + File.separator + STATE_FILE_NAME);
+            while (DeployResultFileUtils.isTfStateFileLocked(path)) {
+                try {
+                    Thread.sleep(1000);
```

Comment:
please consider using Awaitality instead of sleep.

Awaitility.await()
            .atMost(timeout)
            .pollInterval(Duration.ofSeconds(1))
            .until(() -> 'whatever condition';